### DISCUSSION TITLE
Managed Properties not returning all data

### DISCUSSION THREAD

**Discussion by ChrisStoneham** [microsoft-search/pnp-modern-search#2755] [Q&A]
I've been attempting to customise a Details List layout on the Search Results web part to pull through additional columns in the results. I've followed [an article I found](https://sharepointmaven.com/how-to-create-a-managed-property-in-sharepoint-online/) about how to add these as Managed Properties, and then I manually added (by typing and hitting Enter) each one to the Selected Properties on the first pane of the web part configuration.

FYI: One of the additional columns is a site column, the other two are RefinableXXX columns that I've mapped in Managed properties to the crawled properties.

I've added these additional three properties to the Manage Columns configuration for the layout options:

<img width="620" alt="image" src="https://user-images.githubusercontent.com/33757550/217489801-24b35231-25ce-4079-8235-1d3134b95fb8.png">
So far, so good. But, in my results only _some_ of the items are returning are returning data in the additional columns within the search web part, even when they contain data in the document library. So it appears the mapping is right, but, I don't understand why the data for only some items is pulling through.

<img width="595" alt="image" src="https://user-images.githubusercontent.com/33757550/217490845-07c0240d-b012-46ba-a670-7257b5d8cd98.png">

Items with no data showing in Review Date and Type **do** mostly have data in the document library, but, it's not returning. Summary isn't fully populated yet (this is a Site column, not the built in one), but, isn't returning the data we do have in there.

Any ideas?

**Comment by kasperbolarsen** [comment]
Hi @ChrisStoneham 
The first step should always be to verify that the data exists in the result from fra API.  I suggest using SharePoint Search Query tool as there are both guides and videos about on the web.
From the result in the tool you can verify both the query,  the managed/ crawled properties and the values.😁

**Reply by ChrisStoneham** [reply]
I've just tried that - appears the Search Query Tool won't accept the path to the specific document library I'm searching as it just gives a 404 and only works if I set it to the site URL, so, I can't recreate the search I'm doing with the web part.

**Reply by ChrisStoneham** [reply]
So, putting that aside for a second - if I search at the site level I can re-create what I see in the web part, some items are returning the managed properties, but, some aren't. Any ideas why this might be?

**Reply by kasperbolarsen** [reply]
There are a number of reasons: 

the item has not be crawled resently ( use https://pnp.github.io/powershell/cmdlets/Get-PnPSearchCrawlLog.html to verify)

The Item has broken permissions and your account can't access it

The list is set to not be indexed

In this case option 1 seems like the most likely reason

**Reply by ChrisStoneham** [reply]
Well, my Crawl Log isn't showing any reference to that document library at all. What I don't quite understand though is that some items have data. Wouldn't they all be blank if the library hadn't been crawled?

I had already tried a Document Library Re-Index as advised by the guidance I'd found regarding creating he Managed Properties, so, I guess that hasn't worked?

**Reply by kasperbolarsen** [reply]
Did you use Get-PnPSearchCrawlLog -filter "theURLofthelistorlibrary" ?

One of the tricky things about search in SPO and MS search is that we can't control the frequency of the search crawler so sometimes new or updated items are available in the search index within 5-10 minutes and sometimes it takes 5-10 hours or more. 
So patience is a must

**Reply by kasperbolarsen** [reply]
BTW are the fields created as site columns in a content type or just list columns?

**Reply by ChrisStoneham** [reply]
OK - I'm getting somewhere now.  Thanks for your help on this.

Yes, I was using Get-PnPSearchCrawlLog -filter "theURLofthelistorlibrary" before, but it was returning an error. It's now returning results. The only difference is that I kicked off a whole site re-index just before, not just a Document Library re-index, so, looks like that's kicked it into life.

To follow up on your second question, as I think this is still relevant:

- I have two columns which are list columns, so, I've mapped those to a RefinableString and a RefinableDate managed property. These are being returned correctly now.
- I have another which is created as a Site column. This one still isn't returning any data.

**Reply by wobba** [reply]
See https://learn.microsoft.com/en-us/microsoft-365/community/how-do-site-columns-become-managed-properties-thus-available-for-search for some column to search info.
